### PR TITLE
Workaround for an unexpected NPE

### PR DESCRIPTION
**Pull request by ununhexium**
[692](https://github.com/sovity/PMO-Software/issues/692)

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [ ] ~~I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.~~
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java [hunk 1]
```diff
@@ -76,26 +78,33 @@ public List<TransferHistoryEntry> getTransferHistoryEntries() {
...
-            if (direction == ContractAgreementDirection.CONSUMING) {
-                transferHistoryEntry.setAssetName(asset.getId());
-            } else {
-                transferHistoryEntry.setAssetName(
-                        StringUtils.isBlank((String) asset.getProperties().get(Prop.Dcterms.TITLE))
-                                ? asset.getId()
-                                : asset.getProperties().get(Prop.Dcterms.TITLE).toString()
-                );
+
+            if (direction.isPresent()) {
```
Comment (efiege) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java:88]: Did we ensure that the Ui can handle null values for the AssetName? Maybe set the assetName to "" if the value is not present. Or pass an error message.

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java [hunk 2]
```diff
@@ -76,26 +78,33 @@ public List<TransferHistoryEntry> getTransferHistoryEntries() {
...
+                    );
+                }
             }
-            transferHistoryEntry.setContractAgreementId(agreement.getId());
-            transferHistoryEntry.setCounterPartyConnectorEndpoint(negotiation.getCounterPartyAddress());
-            transferHistoryEntry.setCounterPartyParticipantId(negotiation.getCounterPartyId());
-            transferHistoryEntry.setCreatedDate(utcMillisToOffsetDateTime(negotiation.getCreatedAt()));
-            transferHistoryEntry.setDirection(direction);
+
+            agreement.ifPresent(it -> transferHistoryEntry.setContractAgreementId(it.getId()));
```
Comment (efiege) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java:100]: Same here

**Review by efiege — CHANGES_REQUESTED** [review]

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java (hunk 1 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java:88]: Valid concern.

Testing with missing data:

image

image


The UI handles it gracefully. Currently not showing anything but returning null lets the UI decide what and how to show the problem.
If this is the only place where this API is used we're good.

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java (hunk 2 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java:100]: Same

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java (hunk 1 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java:88]: Also, no warning nor error was shown in the browser's console.

**Review by efiege — APPROVED** [review]

**Pull request merged**